src/jimple-frontend/AST/jimple_file.cpp

```diff
@@ -0,0 +1,151 @@
+#include <map>
+#include <fstream>
+#include <jimple-frontend/AST/jimple_file.h>
+#include <util/std_code.h>
+#include <util/expr_util.h>
+
+// (De)-Serialization helpers (from jimple_ast.h)
+void to_json(json &, const jimple_ast &)
+{
+  // Don't care
+}
+
+void from_json(const json &j, jimple_ast &p)
+{
+  p.from_json(j);
+}
+
+std::string jimple_file::to_string() const
+{
+  std::ostringstream oss;
+  oss << "Jimple File:\n"
+      << "\t"
+      << "Name: " << this->class_name << "\n\t"
+      << "Mode: " << to_string(this->mode) << "\n\t"
+      << "Extends: " << this->extends << "\n\t"
+      << "Implements: " << this->implements << "\n\t" << this->m.to_string();
+
+  oss << "\n\n";
+  for(auto &x : body)
+  {
+    oss << x.to_string();
+    oss << "\n\n";
+  }
+
+  return oss.str();
+}
+
+void jimple_file::from_json(const json &j)
+{
+  // Get ClassName
+  j.at("name").get_to(this->class_name);
+
+  std::string t;
+  j.at("object").get_to(t);
+  this->mode = from_string(t);
+
+  try
```

Comment:
Perhaps, these try-catch blocks can be replaced by if-else blocks for better performance.
`if (j.contains("implements"))`

If json doesn't have this key, it will return false.